compiler/noirc_evaluator/src/ssa/ir/cfg.rs

```diff
@@ -89,10 +89,6 @@ impl ControlFlowGraph {
     /// Add a directed edge making `from` a predecessor of `to`.
     fn add_edge(&mut self, from: BasicBlockId, to: BasicBlockId) {
         let predecessor_node = self.data.entry(from).or_default();
-        assert!(
```

Comment:
Now that we have reversed CFGs if a node has many predecessors, you can have multiple successors upon reversing the CFG. If we want to keep this assert I could potentially add a reversed status flag to the CFG.

Reply:
Yes, I would prefer to keep this assert somehow

Reply:
I added a `reversed` flag to keep this assert